packages/radix-ui-themes/src/components/base-button.tsx

```diff
@@ -41,10 +47,38 @@ const BaseButton = React.forwardRef<BaseButtonElement, BaseButtonProps>((props,
         className,
         withBreakpoints(size, 'rt-r-size'),
         `rt-variant-${variant}`,
-        { 'rt-high-contrast': highContrast },
+        {
+          'rt-high-contrast': highContrast,
+          'rt-loading': loading,
+        },
         withMarginProps(marginProps)
       )}
-    />
+      disabled={disabled}
+    >
+      {loading ? (
+        <>
+          {/**
+           * We need a wrapper to set `visibility: hidden` to hide the button content whilst we show the `Spinner`.
+           * The button is a flex container with a `gap`, so we use `display: contents` to ensure the correct flex layout.
+           *
+           * However, `display: contents` removes the content from the accessibility tree in some browsers,
+           * so we force remove it with `aria-hidden` and re-add it in the tree with `VisuallyHidden`
+           */}
+          <span style={{ display: 'contents', visibility: 'hidden' }} aria-hidden>
+            {children}
+          </span>
+          <VisuallyHidden>{children}</VisuallyHidden>
+
+          <Flex asChild align="center" justify="center" position="absolute" inset="0">
+            <span>
```

Comment:
Is this not to avoid putting a div into buttons?

Reply:
I'm not sure if it's strictly needed, but my legacy brains takes over 😀 
It kinda just feels cleaner with `span` when looking at the DOM too.

Reply:
Yeah I think it's one of the reasons we should have as="span"

Reply:
Technically spans in buttons are considered wrong afair